rust/headless-client/src/uptime_lib/mod.rs

```diff
@@ -0,0 +1,85 @@
+// Manually vendored from https://github.com/itchyny/uptime-rs/blob/e6e31f4aa69b057d0610c20f8f2b5f8f0724decb/src/lib.rs and https://github.com/itchyny/uptime-rs/blob/e6e31f4aa69b057d0610c20f8f2b5f8f0724decb/tests/lib.rs
+
+use std::time::Duration;
+
+#[derive(Debug, thiserror::Error)]
+#[cfg_attr(target_os = "windows", allow(unused))]
+pub enum Error {
+    #[cfg(target_os = "linux")]
+    #[error("sysinfo failed")]
+    Sysinfo,
+
+    #[cfg(any(
+        target_os = "macos",
+        target_os = "freebsd",
+        target_os = "openbsd",
+        target_os = "netbsd"
+    ))]
+    #[error("sysctl failed")]
+    Sysctl,
+
+    #[cfg(any(
+        target_os = "macos",
+        target_os = "freebsd",
+        target_os = "openbsd",
+        target_os = "netbsd"
+    ))]
+    #[error(transparent)]
+    SystemTime(#[from] std::time::SystemTimeError),
+}
+
+#[cfg(target_os = "linux")]
+pub fn get() -> Result<Duration, Error> {
+    let mut info: libc::sysinfo = unsafe { std::mem::zeroed() };
+    let ret = unsafe { libc::sysinfo(&mut info) };
+    if ret == 0 {
+        Ok(Duration::from_secs(info.uptime as u64))
+    } else {
+        Err(Error::Sysinfo)
+    }
+}
+
+#[cfg(any(
+    target_os = "macos",
+    target_os = "freebsd",
+    target_os = "openbsd",
+    target_os = "netbsd"
+))]
+pub fn get() -> Result<Duration, Error> {
```

Comment:
We don't support MacOS anywhere else in the headless-client. Can we make this a stub too?

Reply:
I was thinking it'd be easier to copy any changes from upstream this way, but yeah we could

Reply:
That would be closer to maintaining a fork. If we vendor, we should treat the code the same as we treat our own code.

Reply:
Yeah I'll remove the macOS code and accept the other changes then. Since the `uptime_lib` crate is so small, it's likely that `windows` will do this silly version breaking thing multiple times per year while the APIs for system uptime aren't going to change much